### PR TITLE
e2e tests for proxy

### PR DESCRIPTION
**Pull request by rajivnathan**
Related issue: https://issues.redhat.com/browse/CRT-1274

Related PR: https://github.com/codeready-toolchain/registration-service/pull/228

Add a test to provision users to the appstudio tier and verify that the proxy API can be used with the user namespaces.

### CODE REVIEW

File: test/e2e/appstudio_test.go
Comment (rajivnathan) [reply test/e2e/appstudio_test.go:1]: Maybe we should have an `appstudio` package within the `e2e` package? Then we can put all our appstudio-specific tests there?

Comment (MatousJobanek) [reply test/e2e/appstudio_test.go:1]: Technically, all e2e tests are used also for AppStudio case - not only Dev Sandbox. 
I would just avoid using the term AppStudio as it could confuse some ppl. I would vote for naming it `proxy_test.go` for now. We can change it later to `AppStudio` if there is any need in using it.

File: test/e2e/appstudio_test.go
Comment (MatousJobanek) [inline test/e2e/appstudio_test.go:143]: it doesn't matter if it's for the first one or the second one because the namespace will be deleted by the cleanup anyway.

File: test/e2e/appstudio_test.go
Comment (rajivnathan) [reply test/e2e/appstudio_test.go]: Renamed to `proxy_test.go` 75080f9ffbcd1671fbb15590b41205ef7a02c5ec

File: test/e2e/appstudio_test.go
Comment (rajivnathan) [reply test/e2e/appstudio_test.go]: Right, I meant to clean this up. Thanks for the reminder. Done: db779c80c2e249cb857c283d72753d5038b50cc7

File: test/e2e/appstudio_test.go
Comment (rajivnathan) [reply test/e2e/appstudio_test.go]: yeah, that would be better. Thanks for the suggestion! Done: 97d147afb97c72beb647e1952bfd5f15076bc1ff

File: test/e2e/appstudio_test.go
Comment (rajivnathan) [reply test/e2e/appstudio_test.go]: 👍  Removed: 5c170c87bf68e6aeacf6a3cb431c1cf417a6b40f

File: testsupport/wait/host.go
Comment (rajivnathan) [reply testsupport/wait/host.go:1298]: I did try to use it initially but it requires an endpoint to return a status OK response but since the proxy doesn't have another endpoint it fails. I can change this when we have the health probe endpoint for the proxy defined. Actually, I'm putting that PR up today so maybe I can hold off on merging this one until after that. So I can switch to use `WaitForRouteToBeAvailable` once it's merged.

File: test/e2e/appstudio_test.go
Comment (rajivnathan) [reply test/e2e/appstudio_test.go]: Hmm, good point. I didn't notice that. But now that you recommended to remove the "delete usersignup and expect all resources to be deleted" step I can verify it for the second user. I added a check to ensure the first user's namespace still exists before trying to create the configmap.

File: testsupport/wait/host.go
Comment (rajivnathan) [reply testsupport/wait/host.go:1298]: The proxy health/readiness probe PR is up https://github.com/codeready-toolchain/registration-service/pull/230

Comment (MatousJobanek) [reply testsupport/wait/host.go:1310]: never mind - it's part of the reg-service deployment and before it gets the proxy URL it also waits until the reg-service is there anyway.

Comment (MatousJobanek) [reply testsupport/wait/host.go:1310]: but the health/readiness probe will help for sure :+1:

File: test/e2e/appstudio_test.go
Comment (MatousJobanek) [reply test/e2e/appstudio_test.go]: cool :+1:

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**
